services/service-runner/app/workers/pull-worker.ts

```diff
@@ -85,31 +90,62 @@ async function _primeService(): Promise<void> {
 /**
  * Requests work items from Harmony
  */
-async function _pullWork(): Promise<{ item?: WorkItemRecord; status?: number; error?: string, maxCmrGranules?: number }> {
-  try {
-    const response = await axiosGetWork
-      .get(workUrl, {
-        params: { serviceID: env.harmonyService, podName: env.myPodName },
-        responseType: 'json',
-        validateStatus(status) {
-          return status === 404 || (status >= 200 && status < 400);
-        },
-      });
+async function _pullWork(): Promise<{ item?: WorkItemRecord; status?: number; error?: string, maxCmrGranules?: number; }> {
+  let retries = 0;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let result: { item?: any, status: number, error?: string, maxCmrGranules?: number; };
+  while (!result && retries <= maxGetWorkRetries) {
```

Comment:
I think we want to make this truly retries `maxGetWorkRetries` times. Currently, it does `maxGetWorkRetries - 1` retries.

Reply:
Looking at this again, I think it actually is correct. `retries <= maxGetWorkRetries` will result in it retrying `maxGetWorkRetires` times. Also, `maxGetWorkRetries == Number.MAX_SAFE_INTEGER`, so it probably doesn't really matter.

Reply:
yeah, that's right.